pycloudlib/azure/util.py

```diff
@@ -0,0 +1,88 @@
+# This file is part of pycloudlib. See LICENSE file for license information.
+"""Azure Util Functions."""
+from azure.common.client_factory import (get_client_from_cli_profile,
+                                         get_client_from_json_dict)
+from knack.util import CLIError
+
+
+def get_client(resource, config_dict):
+    """Get azure client based on the give resource.
+
+    This method will first verify if we can get the client
+    by using the information provided on the login account
+    of the user machine. If the user is not logged into Azure,
+    we will try to get the client from the ids given by the
+    user to this class.
+
+    Args:
+        resource: Azure Resource, An Azure resource that we want to get
+                    a client for.
+        config_dict: dict, Id parameters passed by the user to this class.
+
+    Returns:
+        The client for the resource passed as parameter.
+
+    """
+    try:
+        client = get_client_from_cli_profile(resource)
+    except CLIError:
+        client = None
```

Comment:
We should just early exit on success and log a debug message on exception. This way we avoid indenting all the rest of the code below when we fall through on error.

```suggestion
        client = get_client_from_cli_profile(resource)
    except CLIError:
        logger.debug("No azure-cli configuration found. Relying on explicit pycloudlib params")
```